lib/color_picker.rb

```diff
@@ -0,0 +1,47 @@
+require "color"
+require "color/rgb/contrast"
+
+class ColorPicker
+  MINIMUM_CONTRAST = 0.4
+
+  def initialize(name: nil)
+    @name = name
+  end
+
+  def color_contrast
+    @color_contrast ||= generate_contrasting_color
+  end
+
+  def color
+    @color ||= generate_random_color
+  end
+
+  def color_from_file
+    @color_from_file ||= Color::RGB.new(*sample_from_file)
```

Comment:
Should these memoized functions instead be ivars set in the constructor + a reader? Typically, memoized readers really want just read data that is set in the constructor.